src/hooks/useActiveHover.ts

```diff
@@ -0,0 +1,50 @@
+"use client"
+
+import React, { useCallback, useEffect, useState } from "react"
+
+import { debounce } from "@src/utils/debounce"
+
+type ReactMouseEventOrNull = React.MouseEvent<HTMLDivElement, MouseEvent> | null
+
+export const useActiveHover = () => {
+  const [isActive, setIsActive] = useState(false)
+  const [event, setEvent] = useState<ReactMouseEventOrNull>(null)
+
+  const handleMouseOver = useCallback(
+    debounce((e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      setIsActive(true)
+      setEvent(e)
```

Comment:
Maybe event should be reset somewhere, or even kept in REF as it is not used for rendering anywhere.

Reply:
Yes, we can use Ref instead, as we already have trigger for lifesycle update as isActive, thus it should be enough.